api/envoy/extensions/geoip_providers/maxmind/v3/maxmind.proto

```diff
@@ -42,6 +42,10 @@ message MaxMindConfig {
   // Database file is expected to have .mmdb extension.
   string isp_db_path = 5 [(validate.rules).string = {pattern: "^$|^.*\\.mmdb$"}];
 
+  // Indicates that the Maxmind geoip provider should read the autonomous_system_number information
+  // from Maxmind ISP database instead of reading it from Maxmind ASN database.
+  bool read_asn_info_from_isp_db = 6;
```

Comment:
so if this is true, you will read from `asn_db_path`?  How did we determine before whether to use `asn_db_path`?

Reply:
`asn_db_path` didn't exist before my previous patch https://github.com/envoyproxy/envoy/pull/38714/files. We were getting the autonomous_system_number from the isp_db_path. The idea is to allow the previous behaviour since there is an overlap of fields in the dbs,

Reply:
I might be missing something, but if `isp_db_path` is populated (i.e. is non-empty) and if `asn_db_path` is not populated (i.e. is empty), can't we use that to determine that we should read ASN info from the ISP DB instead of the ASN DB?

Reply:
Instead of adding new api field, could we use the following logic?
* if only asn_db is configured, read asn number from it.
* if only isp_db is configured, read asn number from it.
* if both dbs are configured, means users have access to both dbs, read asn number from asn_db.
This should cover all the cases?